libs/client.go

```diff
@@ -68,6 +75,73 @@ func NewClient() *Client {
 	return &client
 }
 
+// SetProxy adds a proxy to the client
+func (c *Client) SetProxy(proxyList []string) {
+list:
+	for _, p := range proxyList {
+		proxyurl, err := url.Parse(p)
+		if err == nil {
+			if err := c.setProxy(proxyurl); err == nil {
+				gologger.Info().Msgf("Using %s proxy %s", proxyurl.Scheme, p)
+				break
+			} else {
+				gologger.Error().Msgf("Error setting proxy %s: %s", p, err)
+			}
+		} else {
+			if fileutil.FileExists(p) {
+				file, err := os.Open(p)
+				if err != nil {
+					continue
+				}
+				defer file.Close()
+				scanner := bufio.NewScanner(file)
+				for scanner.Scan() {
+					proxy := scanner.Text()
+					if strings.TrimSpace(proxy) == "" {
+						continue
+					}
+					proxyurl, err := url.Parse(proxy)
```

Comment:
I think this code can be moved into a proper function since it's repeated at line 82